sopel/test_tools.py

```diff
@@ -47,7 +47,9 @@ def __init__(self, nick, admin=False, owner=False):
         self.nick = nick
         self.user = "sopel"
 
-        self.channels = ["#channel"]
+        channel = sopel.tools.Identifier("#Sopel")
```

Comment:
I think I had changed it to `#Sopel` in my other PR to be in line with an example channel in `test_module.py` and possibly others. The only non-module use of `#channel` is in test_tools.py L124, it's not obvious what that affects but we could change that one too.
I don't think MockSopel.channels was used meaningfully, since lists and dicts aren't really interchangeable, and all tests pass.

Reply:
And it's a dict in the actual implementation, so that's an important change actually.

Reply:
Hey, if the tests pass, I don't complain. It's more out of curiosity, because that's the kind of thing @dgw see all the time and ask about - for good reason.

Here, it looks all good.

Reply:
> that's the kind of thing @dgw see all the time and ask about

Uh-oh, I got people learning from and imitating me now… better hide! :D